server/s3_put_versioning_body.cc

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#include <cctype>
+#include <algorithm>
+#include <libxml/parser.h>
+#include <libxml/xmlmemory.h>
+
+#include <regex>
+#include "s3_log.h"
+#include "s3_put_versioning_body.h"
+
+S3PutVersioningBody::S3PutVersioningBody(std::string &xml, std::string &request)
+    : xml_content(xml), request_id(request), is_valid(false) {
+
+  s3_log(S3_LOG_DEBUG, request_id, "%s Ctor\n", __func__);
+  if (!xml.empty()) {
+    parse_and_validate();
+  }
+}
+
+bool S3PutVersioningBody::isOK() { return is_valid; }
+
+bool S3PutVersioningBody::parse_and_validate() {
+  /* Sample body:
+  <VersioningConfiguration>
+   <MfaDelete>string</MfaDelete>
```

Comment:
We need to handle case where if MfaDelete is present we return OperationNotSupported.

Reply:
Done Changes as per suggestion.